canvas_modules/common-canvas/__tests__/common-properties/actions/image-action-test.js

```diff
@@ -96,86 +108,92 @@ describe("action-image renders correctly", () => {
 		controller.updateActionState(actionStateId, "disabled");
-		const wrapper = mount(
+		const wrapper = render(
 			<Provider store={controller.getStore()}>
 				<ActionImage
 					action={action}
 					controller={controller}
 				/>
 			</Provider>
 		);
-		const imageWrapper = wrapper.find(".properties-action-image");
-		expect(imageWrapper.hasClass("disabled")).to.equal(true);
+		const image = wrapper.getByRole("img");
+		const imageWrapper = image.parentElement.parentElement.parentElement.parentElement;
+		expect(imageWrapper.className.includes("disabled")).to.equal(true);
 	});
 	it("action image renders when hidden", () => {
 		controller.updateActionState(actionStateId, "hidden");
-		const wrapper = mount(
+		const wrapper = render(
 			<Provider store={controller.getStore()}>
 				<ActionImage
 					action={action}
 					controller={controller}
 				/>
 			</Provider>
 		);
-		const imageWrapper = wrapper.find(".properties-action-image");
-		expect(imageWrapper.hasClass("hide")).to.equal(true);
+		const image = wrapper.getByRole("img");
+		const imageWrapper = image.parentElement.parentElement.parentElement.parentElement;
+		expect(imageWrapper.className.includes("hide")).to.equal(true);
 	});
 	it("action image renders tooltip", () => {
-		const wrapper = mount(
+		const wrapper = render(
 			<Provider store={controller.getStore()}>
 				<ActionImage
 					action={action}
 					controller={controller}
 				/>
 			</Provider>
 		);
-		const tooltip = wrapper.find("div.tooltipContainer");
+		const tooltip = wrapper.getAllByText("Click to rotate through moon phases.");
+		const tooltipWrapper = tooltip[0].parentElement;
 		expect(tooltip).to.have.length(1);
-		expect(tooltip.text()).to.equal("Click to rotate through moon phases.");
+		expect(tooltipWrapper.className).to.equal("tooltipContainer");
+		expect(tooltip[0].textContent).to.equal("Click to rotate through moon phases.");
 	});
 });
 
 describe("actions using paramDef", () => {
 	let wrapper;
 	let renderedObject;
 	beforeEach(() => {
-		renderedObject = propertyUtils.flyoutEditorForm(ACTION_PARAMDEF);
+		renderedObject = propertyUtils.flyoutEditorFormRender(ACTION_PARAMDEF);
 		wrapper = renderedObject.wrapper;
 	});
 
 	it("should fire action when image clicked", (done) => {
-		renderedObject = propertyUtils.flyoutEditorForm(ACTION_PARAMDEF, null, { actionHandler: callback }, { appData: appData });
+		renderedObject = propertyUtils.flyoutEditorFormRender(ACTION_PARAMDEF, null, { actionHandler: callback }, { appData: appData });
 		wrapper = renderedObject.wrapper;
 		function callback(id, inAppData, data) {
 			expect(id).to.equal("moon");
 			expect(inAppData).to.eql(appData);
 			expect(data.parameter_ref).to.equal("moon_phase");
 			done();
 		}
-		const image = wrapper.find("div[data-id='properties-ctrl-moon_phase']").find("div[data-id='moon'] img");
-		image.simulate("click");
+		const image = wrapper.getAllByRole("img")[21];
+		fireEvent.click(image);
 	});
```

Comment:
Is it possible to still use data-id / data-test-id wherever we used to check ?
Something like below.
```
const { container } = wrapper;
const div = container.querySelector("div[data-id='']");
const button = within(div).getByRole('button');
fireEvent.click(button);
```